testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/interceptor/register/RegisterInterceptorViaMetaFileTest.java

```diff
@@ -0,0 +1,85 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 2110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.integration.ejb.interceptor.register;
+
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+/**
+ * @author Jiri Bilek
+ * jbilek@redhat.com on 12/03/18.
+ * Test for WFLY-9844
+ *
+ * EJB invocation for Remote interface fails when Client Interceptor
+ * registered via META-INF/services/org.jboss.ejb.client.EJBClientInterceptor
+ */
+@RunWith(Arquillian.class)
+public class RegisterInterceptorViaMetaFileTest {
+   private static final String ARCHIVE_NAME = "test-register-interceptor";
+   @ArquillianResource
+   Deployer deployer;
+
+   @Deployment(name = "test-register-interceptor", testable = false, managed = false)
+   public static Archive<?> deploy() {
+      JavaArchive jar = ShrinkWrap.create(JavaArchive.class, ARCHIVE_NAME + ".jar");
+      jar.addPackage(TestRemote.class.getPackage());
+      jar.addPackage(TestSingleton.class.getPackage());
+      jar.addPackage(TestSLSB.class.getPackage());
+      jar.addAsManifestResource(
+            RegisterInterceptorViaMetaFileTest.class.getPackage(),
+            "org.jboss.ejb.client.EJBClientInterceptor",
+            "services/org.jboss.ejb.client.EJBClientInterceptor");
+      return jar;
+   }
+
+   @Test
+   public void testInvalidTransactionAttributeWarnLogged() {
+      PrintStream oldOut = System.out;
+      ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      try {
+         System.setOut(new PrintStream(baos));
+         deployer.deploy(ARCHIVE_NAME);
+         try {
+            System.setOut(oldOut);
+            String output = new String(baos.toByteArray());
+            Assert.assertTrue("TestSingleton did not start with work", output.contains("Testing string: singleton starting with work"));
+            Assert.assertTrue("@EJB injection does not work when ClientInterceptor is registered via META-INF/services", output.contains("Testing string: SLSB successfully invoked"));
+            Assert.assertTrue("TestSingleton is not able to finish his work", output.contains("Testing string: information about ClientInterceptor"));
```

Comment:
This is not how you write a test. Just put your test logic in the test itself. Don't put the test logic in a @PostConstruct method, log the result, and then have the test attempt to verify the logs.